src/coreclr/gc/handletableconstants.h

```diff
@@ -14,7 +14,7 @@
 #endif
 
 #define INITIAL_HANDLE_TABLE_ARRAY_SIZE 10
-#define HANDLE_MAX_INTERNAL_TYPES       12
+#define HANDLE_MAX_INTERNAL_TYPES       16
```

Comment:
Are there perf consequences from bumping this limit? Should we start recycling deprecated HDNTYPE values (e.g. HNDTYPE_ASYNCPINNED)?

Reply:
just need to be careful that recycling wouldnt cause backcompat issues for standalone gc.